File: Java/ArrayString/30_BoatsTosavePeople.java
public int numRescueBoats(int[] arr, int limit) {
        Arrays.sort(arr);
        int n = arr.length, si = 0, ei = n-1;
        int usedBoat =0;
        while(si < ei){
            int wt = arr[si] + arr[ei];
            if(wt <= limit) si++;
            ei--;
            usedBoat++;
            
        }
        if(si==ei) usedBoat++;
        return usedBoat;
    }

File: C++/ArrayString/30_BoatsTosavePeople.cpp
#include <bits/stdc++.h>
using namespace std;

int numRescueBoats(vector<int> &people, int limit)
{
    sort(people.begin(), people.end());
    int n = people.size();
    int si = 0, ei = n - 1;
    int count = 0;
    while (si < ei)
    {
        if (people[si] + people[ei] > limit)
            ei--;
        else
        {
            si++, ei--;
        }
        count++;
    }
    if (si == ei)
        count++;

    return count;
}